ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/flow_tracking.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+from functools import cached_property
+from typing import TYPE_CHECKING
+
+from ansible_collections.arista.avd.plugins.plugin_utils.errors import AristaAvdError
+
+# from ansible_collections.arista.avd.plugins.filter.natural_sort import natural_sort
+# from ansible_collections.arista.avd.plugins.plugin_utils.errors import AristaAvdError, AristaAvdMissingVariableError
+# from ansible_collections.arista.avd.plugins.plugin_utils.strip_empties import strip_empties_from_dict
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import get
+
+if TYPE_CHECKING:
+    from .shared_utils import SharedUtils
+
+
+class FlowTrackingMixin:
+    """
+    Mixin Class providing a subset of SharedUtils
+    Class should only be used as Mixin to the SharedUtils class
+    Using type-hint on self to get proper type-hints on attributes across all Mixins.
+    """
+
+    @cached_property
+    def flow_tracker_name(self: SharedUtils) -> str:
+        """
+        Return the name of the flow tracking object
+        Used in both network services, underlay and overlay python modules.
+        """
+        return get(self.hostvars, "flow_tracking_settings.flow_tracker_name", default="FLOW-TRACKER")
+
+    @cached_property
+    def fabric_flow_tracking(self: SharedUtils):
+        return get(self.hostvars, "fabric_flow_tracking", default={})
+
+    def resolve_flow_tracker_name(self: SharedUtils, settings: dict | None, context_path: str):
+        if settings is None:
+            return None, None, None
+
+        trackerEnabled = settings.get("enabled")
+        trackerType = settings.get("type")
+        trackerName = settings.get("name")
+
+        raiseError = None
+        if trackerType and not trackerName:
+            raiseError = "type"
+        if trackerName and not trackerType:
+            raiseError = "name"
+        if raiseError:
+            raise AristaAvdError(f"Both type and name need to be configured for flow tracker but only '{raiseError}'" f" is configured for {context_path}.")
+        return trackerEnabled, trackerType, trackerName
+
+    def fabric_flow_tracking_data_model(self: SharedUtils, data_model: str) -> dict | None:
+        default = False
+        if data_model == "dps_interfaces":
+            default = True
+
+        flow_tracking = self.fabric_flow_tracking.get(data_model)
+
+        trackerEnabled, trackerType, trackerName = self.resolve_flow_tracker_name(flow_tracking, "")
+
+        return {
+            "enabled": trackerEnabled or default,
```

Comment:
Hmmm for dps interfce, if `trackerEnabled` is set to `False`, this will be `False or True = True` which is not what you want